modules/creating-a-custom-image-from-your-own-image.adoc

```diff
@@ -0,0 +1,160 @@
+
+* Avoid using the `yum update` command. 
+
+** If you start from an image that is constantly updated, such as `ubi9/python-39` from the Red Hat Catalog, you might not need to use the yum update command. This command fetches new metadata, updates files that might not have impact, and increases the notebook image size.
+
+** Rebuild your images from scratch. Avoid running a `yum update` command on a previous version.
+
+* Group `RUN` commands. Chain your commands by adding `&& \` at the end of each line.
+
+* If you must compile code (such as a library or an application) to include in your custom image, implement multi-stage builds so that you avoid including the build artifacts in your final image. That is, compile the library or application in an intermediate image and then copy the result to your final image, leaving behind build artifacts that you do not want included.
+
+*Setting access to files and directories*
+
+* Set the ownership of files and folders to `1001:0` (user "default", group "0"), for example:
++
+----
+COPY --chown=1001:0 os-packages.txt ./
+----
++
+On OpenShift, every container is in a standard namespace (unless you modify security). The container runs with a user that has a random user ID (uid) and with a group ID (gid) of `0`. Therefore, all folders that you want to write to - and all the files you want to (temporarily) modify - in your image must be accessible by the user that has the random user ID (uid).
+Alternatively, you can set access to any user, as shown in the following example:
++
+----
+COPY --chown=775:0 os-packages.txt ./
+----
+
+* Build your image with `/opt/app-root/src` as the default location for the data that you want persisted, for example:
++
+----
+WORKDIR /opt/app-root/src
+----
++
+When a user launches a notebook from the {productname-short}*Applications* -> *Enabled* page, the "personal" volume of the user is mounted at `/opt/app-root/src`. Because this location is not configurable, when you build your custom image, you must specify this default location for persisted data.
+
+* Fix permissions to support PIP (the package manager for Python packages) in OpenShift environments. Add the following command to your custom image:
++
+----
+chmod -R g+w /opt/app-root/lib/python3.9/site-packages && \
+   fix-permissions /opt/app-root -P
+----
+
+* A service within your notebook image must answer at `${NB_PREFIX}/api`, otherwise the OpenShift liveness/readiness probes fail and delete the pod for the workbench image.
++
+The `NB_PREFIX` environment variable specifies the URL path where the container is expected to be listening.
++
+The following is an example of an Nginx configuration:
++
+----
+location = ${NB_PREFIX}/api {
+	return 302  /health;
+	access_log  off;
+}
+----
+
+* For idle culling to work, the `${NB_PREFIX}/api/kernels` URL must return a specifically-formatted JSON payload, as shown in the following example:
```

Comment:
Can delete "as shown in the following example" since you have line 95 (or vice versa).